### PR TITLE
Proposed changes to Repository filtering doc

### PR DESCRIPTION
**Pull request by alongosz**
Suggested changes to #1099.

Main points:
* Moved it to a dedicated page. Filtering is NOT search, it actually uses Query Criteria, but they're referenced everywhere as Search Criteria, which is probably a cause of this misconception. As a compromise I've mentioned Filtering on Search page, so people looking for search can find it there as well. Change is in a separate commit, so can be removed if it's really "no-go".
* Also separately, updated main pages for Criteria and Sort Clauses respectively, to include Repository Filtering. For that purpose instead of saying "Search type", I'm saying "Supported by" and changing "Content and Location" to "Content & Location Search"
* `IsMainLocation` Criterion is actually supported, only Sort clause is not, as it mostly does not make semantic sense.
*  Suggested minor changes to the #1099 doc, can be applied regardless of the above points via 
f4b103b.

### CODE REVIEW

**Comment by alongosz** [comment]
Updated according to internal sync.

File: docs/api/public_php_api_search.md [hunk 1]
```diff
@@ -98,7 +102,7 @@ Filtering differs from search. It does not use the `SearchService` and is not ba
 For example, the following command lists all Content items under the specified parent Location
 and sorts them in descending order by name:
 
-``` php hl_lines="13 14 15 16"
+``` php hl_lines="15-18"
```
Comment (DominikaK) [inline docs/api/public_php_api_search.md:105]:
```suggestion
``` php hl_lines="15 16 17 18"
```

File: docs/api/public_php_api_search.md [hunk 2]
```diff
@@ -120,14 +126,62 @@ class FilterCommand extends Command
...
+            $output->writeln($location->getContent()->getName());
+        }
+
+        return self::SUCCESS;
     }
 }
 ```
 
+Notice that the total number of items is retrieved differently depending on whether we're dealing
+with `ContentList` or `LocationList`.
```
Comment (DominikaK) [inline docs/api/public_php_api_search.md:173]:
```suggestion
Notice that the total number of items is retrieved differently for `ContentList` and `LocationList`. 
```

File: docs/api/public_php_api_search.md [hunk 3]
```diff
@@ -120,14 +126,62 @@ class FilterCommand extends Command
...
     }
 }
 ```
 
+Notice that the total number of items is retrieved differently depending on whether we're dealing
+with `ContentList` or `LocationList`. 
+
+!!! caution
+
+    The total count is **a total number of matched items**, regardless of pagination settings.
```
Comment (DominikaK) [inline docs/api/public_php_api_search.md:177]:
```suggestion
    The total count is the total number of matched items, regardless of pagination settings.
```

File: docs/api/public_php_api_search.md [hunk 4]
```diff
@@ -120,14 +126,62 @@ class FilterCommand extends Command
...
+    
+!!! tip "Repository Filtering is SiteAccess-aware."
```
Comment (DominikaK) [inline docs/api/public_php_api_search.md:179]:
```suggestion
!!! tip "Repository filtering is SiteAccess-aware"
```

File: docs/api/public_php_api_search.md [hunk 5]
```diff
@@ -120,14 +126,62 @@ class FilterCommand extends Command
...
+
+    Repository Filtering is SiteAccess-aware, which means that the second argument of its respective
+    `find` methods can be skipped. In that case languages from a current context are injected and
+    added as a LanguageCode Criterion filter.
```
Comment (DominikaK) [inline docs/api/public_php_api_search.md:183]:
```suggestion
    Repository filtering is SiteAccess-aware, which means you can skip the second argument the `find` methods.
    In that case languages from a current context are injected and added as a LanguageCode Criterion filter. 
```

File: docs/api/public_php_api_search.md [hunk 6]
```diff
@@ -152,8 +206,18 @@ $filter
...
+    and Sort Clauses implementing [`FilteringSortClause`](https://github.com/ezsystems/ezplatform-kernel/blob/v1.1.0/eZ/Publish/SPI/Repository/Values/Filter/FilteringSortClause.php)
+    are supported.
+
     See to [Search Criteria](../guide/search/search_criteria_reference.md)
     and [Sort Clause reference](../guide/search/sort_clause_reference.md) for details.
+    
+!!! tip
+
+    It's recommended to use an IDE capable of recognizing type hints when working with Repository Filtering.
+    In case of using unsupported Criterion or Sort Clause, such IDE will immediately hint an issue.
```
Comment (DominikaK) [inline docs/api/public_php_api_search.md:220]:
```suggestion
    It is recommended to use an IDE that can recognize type hints when working with Repository Filtering.
    If you try to use an unsupported Criterion or Sort Clause, the IDE will indicate an issue. 
```

**Pull request merged**